pkcs11/src/api/decrypt.rs

```diff
@@ -19,7 +19,12 @@ pub extern "C" fn C_DecryptInit(
 
     trace!("C_DecryptInit() mech: {:?}", unsafe { *pMechanism });
 
-    let raw_mech = unsafe { CkRawMechanism::from_raw_ptr_unchecked(pMechanism) };
+    let raw_mech = match unsafe { CkRawMechanism::from_raw_ptr(pMechanism) } {
+        Some(mech) => mech,
+        None => {
+            return CKR_ARGUMENTS_BAD;
+        }
+    };
```

Comment:
nit (also applies to other functions with a similar pattern): You could replace the `is_null` call with this block to avoid the duplicate check.